bin-src/lib/getStorybookInfo.ts

```diff
@@ -1,138 +1,25 @@
-      // If we aren't going to invoke the Storybook CLI later, we can exit early.
-      // Note that `version` can be a semver range in this case.
-      return { viewLayer, version };
-    }
-
-    // Verify that the viewlayer package is actually present in node_modules.
-    return Promise.race([
-      resolvePackageJson(pkg)
-        .then((json) => ({ viewLayer, version: json.version }))
-        .catch(() => Promise.reject(new Error(noViewLayerPackage(pkg)))),
-      timeout(10000),
-    ]);
-  }
-
-  if (!options.interactive) {
-    log.info(`No viewlayer package listed in dependencies. Checking transitive dependencies.`);
-  }
-
-  // We might have a transitive dependency (e.g. through `@nuxtjs/storybook` which depends on
-  // `@storybook/vue`). In this case we look for any viewlayer package present in node_modules,
-  // and return the first one we find.
-  return Promise.race([
-    raceFulfilled(
-      Object.entries(viewLayers).map(async ([key, value]) => {
-        const json = await resolvePackageJson(key);
-        return { viewLayer: value, version: json.version };
-      })
-    ).catch(() => Promise.reject(new Error(noViewLayerPackage(pkg)))),
-    timeout(10000),
-  ]);
-};
-
-const findAddons = async ({ packageJson }) => ({
-  addons: Object.entries(supportedAddons)
-    .map(([pkg, name]) => {
-      const [dep, devDep, peerDep] = findDependency(packageJson, ([key]) => key === pkg);
-      const [packageName, packageVersion] = dep || devDep || peerDep || [];
-      return packageName && packageVersion && { name, packageName, packageVersion };
-    })
-    .filter(Boolean),
-});
+import path from 'path';
 
-const findConfigFlags = async ({ options, packageJson }) => {
-  const { scripts = {} } = packageJson;
-  if (!options.buildScriptName || !scripts[options.buildScriptName]) return {};
-
-  const { flags } = meow({
-    argv: parseArgsStringToArgv(scripts[options.buildScriptName]),
-    flags: {
-      configDir: { type: 'string', alias: 'c' },
-      staticDir: { type: 'string', alias: 's' },
-    },
-  });
+import { Context } from '../types';
+import { getStorybookMetadataFromProjectJson } from './getPrebuiltStorybookMetadata';
 
-  return {
-    configDir: flags.configDir,
-    staticDir: flags.staticDir && flags.staticDir.split(','),
-  };
-};
+import { getStorybookMetadata } from './getStorybookMetadata';
 
 export default async function getStorybookInfo(
   ctx: Context
 ): Promise<Partial<Context['storybook']>> {
   try {
-    const info = await Promise.all([findAddons(ctx), findConfigFlags(ctx), findViewlayer(ctx)]);
-    return info.reduce((acc, obj) => Object.assign(acc, obj), {});
+    if (ctx.options.storybookBuildDir) {
+      const projectJsonPath = path.resolve(ctx.options.storybookBuildDir, 'project.json');
+      // This test makes sure we fall through if the file does not exist.
+      if (fs.pathExistsSync(projectJsonPath)) {
+        return await getStorybookMetadataFromProjectJson(projectJsonPath);
```

Comment:
Maybe add a comment to clarify `await` is required here in order for the `catch` block to apply to the result of this promise (typically `return await` is a mistake).

Reply:
Comment added!